substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1843,14 +1875,44 @@ impl<T: Config> Pallet<T> {
 		Self::check_exposures()?;
 		Self::check_paged_exposures()?;
 		Self::check_ledgers()?;
+		Self::check_locks()?;
 		Self::check_count()
 	}
 
+	/// Invariants:
+	///
+	/// * Staking locked funds for every bonded stash should be the same as its ledger's total.
+	///
+	/// NOTE: some of these checks result in warnings only. Once
+	/// <https://github.com/paritytech/polkadot-sdk/issues/3245> is resolved, turn warns into check
+	/// failures.
+	fn check_locks() -> Result<(), TryRuntimeError> {
+		for (_, controller) in <Bonded<T>>::iter() {
```

Comment:
why are you not using your `inspect_bond_state` here?

Reply:
```
Bonded::<T>::iter().all(|(stash, _)| Self::inspect_bond_state(stash).is_ok())
```

Reply:
I refactored a bit the try state checks and added the `inspect_bond_state` for each bonded stash.